api/src/main/openapi/paths/workflow-states__token_.yaml

```
@@ -0,0 +1,42 @@
+# This file is part of Dependency-Track.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (c) OWASP Foundation. All Rights Reserved.
+get:
+  operationId: getWorkflowStates
+  summary: Retrieves workflow states associated with the token received from bom upload.
+  description: Requires permission <strong>BOM_UPLOAD</strong>
+  tags:
+    - Workflow
+  parameters:
+    - name: uuid
+      in: path
+      description: The UUID of the token to query
+      required: true
+      schema:
+        type: string
+  responses:
+    "200":
+      description: A list of workflow states
+      content:
+        application/json:
+          schema:
+            $ref: "../components/schemas/list-workflow-states-response.yaml"
+    "401":
+      $ref: "../components/responses/generic-unauthorized-error.yaml"
```

We respond with a 403 if the client is missing the necessary permissions:

```suggestion
    "401":
      $ref: "../components/responses/generic-unauthorized-error.yaml"
    "403":
      $ref: "../components/responses/generic-forbidden-error.yaml"
```